test/features/errors/notfound.js

```diff
@@ -66,4 +70,95 @@ describe('404 handling', function() {
             assert.contentType(e, 'application/problem+json');
         });
     });
+    it('should return 404 on deleted revision', function() {
+        return preq.get({ uri: server.config.bucketURL + '/revision/' + revPageDeleted })
+            .then(function() {
+                throw new Error('404 should be returned')
+            })
+            .catch(function(e) {
+                assert.deepEqual(e.status, 404);
+                assert.contentType(e, 'application/problem+json');
+            })
+    });
+    it('should set page_deleted on deleted page', function() {
+        var apiURI = server.config
+            .conf.templates['wmf-sys-1.0.0']
+            .paths['/{module:action}']['x-modules'][0].options.apiURI;
+        var title = 'TestingTitle';
+        var revision = 12345;
+
+        nock.enableNetConnect();
```

Comment:
We probably need to make sure that nock either only catches the specific POST we are interested in, or that it is disabled right after the test has run. The former would still be preferable, imho, as it would also work if there are multiple API requests happening as part of this test later on.